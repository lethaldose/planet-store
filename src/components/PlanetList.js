import React, { Component } from 'react'
import Grid from '@material-ui/core/Grid';
import TextField from '@material-ui/core/TextField';
import Planet from '../components/Planet'

const dummyPlanets = [{
  name: 'Mars',
  description: 'The best planet',
  imageUrl: 'http://image-url',
  price: {
    amount: '111.40', currency: 'AUD'
  }
}];

class PlanetList extends Component {
    state = {
        planets: [...dummyPlanets],
        searchString: ''
    }
    // constructor() {
    //   super()
    // }
    getPlanets = () => {
      console.log('getPlanets');
      console.log(dummyPlanets);
      this.setState({planets: [...dummyPlanets]})
    }
    onSearchInputChange = (event) => {
        console.log("Search changed ..." + event.target.value)
        if (event.target.value) {
            this.setState({searchString: event.target.value})
        } else {
            this.setState({searchString: ''})
        }
        this.getPlanets()
    }
    render() {
        return (
            <div>
            { this.state.planets ? (
                <div>
                <TextField style={{padding: 24}}
                id="searchInput"
                placeholder="Search for Planet"
                margin="normal"
                onChange={this.onSearchInputChange}
                />
                <Grid container spacing={24} style={{padding: 24}}>
                { this.state.planets.map(planet => (
                    <Grid item key={planet.name} xs={12} sm={6} lg={4} xl={3}>
                    <Planet planet={planet} />
                    </Grid>
                    ))}
                </Grid>
                </div>
                ) : "No planets found" }
            </div>
            )
    }
}
export default PlanetList;