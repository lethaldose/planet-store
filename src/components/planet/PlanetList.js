import React, { Component } from 'react';
import { connect } from 'react-redux';
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import TextField from '@material-ui/core/TextField';
import Planet from './Planet';
import { fetchPlanets, filterPlanets } from './planetActions';

export class PlanetList extends Component {
  componentWillMount() {
    this.props.fetchPlanets();
  }

  onSearchInputChange = event => {
    this.props.filterPlanets(event.target.value);
  };

  render() {
    const { planets } = this.props;
    return (
      <div>
        {planets && planets.length > 0 ? (
          <div>
            <TextField
              style={{ padding: 24 }}
              id="searchInput"
              placeholder="Search for Planet"
              margin="normal"
              onChange={this.onSearchInputChange}
            />
            <Grid container spacing={24} style={{ padding: 24 }}>
              {planets.map(planet => (
                <Grid item key={planet.name} xs={12} sm={6} lg={4} xl={3}>
                  <Planet planet={planet} />
                </Grid>
              ))}
            </Grid>
          </div>
        ) : (
          <Grid item style={{ padding: 24 }}>
            <Typography color="error" variant="h5" component="h2">
              No planets available!
            </Typography>
          </Grid>
        )}
      </div>
    );
  }
}

const mapStateToProps = state => {
  const { planets } = state.planetReducer;
  return {
    planets
  };
};

const mapDispatchToProps = {
  fetchPlanets,
  filterPlanets
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(PlanetList);
