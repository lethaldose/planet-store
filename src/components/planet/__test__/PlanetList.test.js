import React from 'react';
import { mount, shallow } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';
import { PlanetList } from '../PlanetList';

describe('PlanetList Component', () => {
  const planet1 = {
    name: 'Mars',
    description: 'The best planet',
    imageUrl: 'https://dummy.jpeg',
    price: {
      amount: '111.40',
      currency: 'AUD'
    },
    area: {
      value: '1231000.12',
      unit: 'sq km'
    },
    radius: {
      value: '12310',
      unit: 'km'
    },
    orbitPeriod: {
      value: '2323',
      unit: 'days'
    },
    gravity: {
      value: '3.23',
      unit: 'm/s2'
    },
    temperature: {
      value: '213.12',
      unit: 'C'
    }
  };

  let defaultProps;

  const getComponent = props => {
    return (
      <BrowserRouter>
        <PlanetList {...props} />
      </BrowserRouter>
    );
  };

  beforeEach(() => {
    defaultProps = {
      planets: [],
      fetchPlanets: jest.fn()
    };
  });

  it('fetches planets', () => {
    shallow(<PlanetList {...defaultProps} />);
    expect(defaultProps.fetchPlanets).toHaveBeenCalled();
  });

  it('render planets', () => {
    const props = { ...defaultProps, planets: [planet1] };
    const wrapper = mount(getComponent(props));
    expect(wrapper.find('div.planet-card').length).toEqual(1);
    expect(wrapper).toMatchSnapshot();
  });

  it('render empty planets message', () => {
    const wrapper = mount(getComponent(defaultProps));
    expect(wrapper.text()).toEqual('No planets available!');
  });
});
