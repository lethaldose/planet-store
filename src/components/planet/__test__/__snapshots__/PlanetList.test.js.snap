// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlanetList Component render planets 1`] = `
<div>
  <div>
    <TextField
      id="searchInput"
      margin="normal"
      onChange={[Function]}
      placeholder="Search for Planet"
      required={false}
      select={false}
      style={
        Object {
          "padding": 24,
        }
      }
      variant="standard"
    />
    <WithStyles(Grid)
      container={true}
      spacing={24}
      style={
        Object {
          "padding": 24,
        }
      }
    >
      <WithStyles(Grid)
        item={true}
        key="Mars"
        lg={4}
        sm={6}
        xl={3}
        xs={12}
      >
        <Planet
          planet={
            Object {
              "description": "The best planet",
              "imageUrl": "https://dummy.jpeg",
              "name": "Mars",
              "price": Object {
                "amount": "111.40",
                "currency": "AUD",
              },
            }
          }
        />
      </WithStyles(Grid)>
    </WithStyles(Grid)>
  </div>
</div>
`;
